apps/src/lib/script-editor/LevelNameInput.jsx

```diff
@@ -0,0 +1,65 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import FontAwesome from '../../templates/FontAwesome';
+
+const styles = {
+  levelSelect: {
+    display: 'inline-block',
+    verticalAlign: 'middle',
+    width: 600
+  },
+  validIcon: {
+    color: 'green',
+    fontSize: 16,
+    marginLeft: 6
+  },
+  invalidIcon: {
+    color: 'red',
+    fontSize: 16,
+    marginLeft: 6
+  }
+};
+
+export default class LevelNameInput extends Component {
+  static propTypes = {
+    onSelectLevel: PropTypes.func.isRequired,
+    levelKeyToIdMap: PropTypes.objectOf(PropTypes.number).isRequired,
```

Comment:
rename to `levelNameToIdMap`? i don't feel very strongly about this, but i think it's slightly more straightforward

Reply:
that sounds better. I don't know why this is called levelKey in some places but levelName does seem more accurate.